substrate/frame/elections-phragmen/src/lib.rs

```diff
@@ -458,6 +468,11 @@ pub mod pallet {
         ) -> DispatchResult {
             let who = ensure_signed(origin)?;
 
+            ensure!(
+                T::CandidateIdentityProvider::has_identity(&who),
```

Comment:
This should check for verified identity, i.e. no. of justifications should be greater than 1 else anyone can create identity and submit candidacy instantly.